cassandra/cluster.py

```diff
@@ -2126,11 +2141,13 @@ def _create_response_future(self, query, parameters, trace, custom_payload, time
 
         if isinstance(query, SimpleStatement):
             query_string = query.query_string
+            statement_keyspace = query.keyspace
```

Comment:
The reason we don't compare here if 
`keyspace = query.keyspace if ProtocolVersion.uses_keyspace_flag(self._protocol_version) else None`
is because if the use creates a statement like 
`SimpleStatement("SELECT * from system.local".format(self.table_name), keyspace="system")` I think a exception should be raised. This breakes the current API, since some applications doing this (may be bugged applications) will now raise an Exception if they are using a lower protocol.